proxy/gateway/nodejs/lib/codec.js

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
```

Comment:
Just pushed some commits that introduce a new document, `proxy/message_format.md`. And I added a comment to `codec.js` to link to this document.